src/BitOasis/Coin/Address/CeloAddress.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace BitOasis\Coin\Address;
+
+use BitOasis\Coin\Cryptocurrency;
+use BitOasis\Coin\CryptocurrencyAddress;
+use BitOasis\Coin\CryptocurrencyNetwork;
+use BitOasis\Coin\Exception\InvalidAddressException;
+use Murich\PhpCryptocurrencyAddressValidation\Validation\ETH as ETHValidator;
+
+/**
+ * @author ahmad.yousef <ahmad.yousef@bitoasis.net>
+ */
+class CeloAddress implements CryptocurrencyAddress {
+
+	/** @var string */
+	protected $address;
+
+	/** @var Cryptocurrency */
+	protected $currency;
+
+	/** @var CryptocurrencyNetwork */
+	protected $cryptocurrencyNetwork;
+
+	/**
+	 * @throws InvalidAddressException
+	 */
+	public function __construct(string $address, Cryptocurrency $currency, CryptocurrencyNetwork $cryptocurrencyNetwork) {
+		if (!$this->isValid($address)) {
+			throw new InvalidAddressException('This is not valid celo address - ' . $address);
+		}
+		$this->address = $address;
+		$this->currency = $currency;
+		$this->cryptocurrencyNetwork = $cryptocurrencyNetwork;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getAddress(): string {
+		return $this->address;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function toString(): string {
+		return $this->address;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function serialize(): string {
+		return $this->address;
+	}
+
+	public function getCurrency(): Cryptocurrency {
+		return $this->currency;
+	}
+
+	public function getNetwork(): CryptocurrencyNetwork {
+		return $this->cryptocurrencyNetwork;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function supportsAdditionalId(): bool {
+		return static::supportsClassAdditionalId();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getAdditionalIdName(): ?string {
+		return static::getClassAdditionalIdName();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function supportsClassAdditionalId(): bool {
+		return false;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function getClassAdditionalIdName(): ?string {
+		return null;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getAdditionalId() {
+		return null;
+	}
+
+	public function equals(CryptocurrencyAddress $address): bool {
+		return $address instanceof static && $this->currency->equals($address->currency) && $this->address === $address->address;
+	}
+
+	/**
+	 * @throws InvalidAddressException
+	 */
+	public static function deserialize($string, Cryptocurrency $cryptocurrency, CryptocurrencyNetwork $cryptocurrencyNetwork): CryptocurrencyAddress {
+		return new static($string, $cryptocurrency, $cryptocurrencyNetwork);
+	}
+
+	public function isValid(string $address): bool {
+		return (new ETHValidator($address))->validate();
```

Comment:
Suggestion: I propose creating a CeloAddressValidator and incorporating the logic from the existing validator into it. Although a Celo address is not an Ethereum address, the underlying logic is compatible and can be effectively utilized.

Reply:
Agree. If there is a new ETH address format, this will likely affect Celo in our case as well without notice.